.github/workflows/docker-nightly-image.yml

```diff
@@ -0,0 +1,76 @@
+name: Docker Nightly Image CI
+run-name: Build ${{ inputs.branch }} - ${{ inputs.user }}
+
+on:
+  schedule:
+    - cron: '15 1 * * *'
+  workflow_dispatch:
+    inputs:
+      repository:
+        description: 'GitHub repository to create image off.'
+        required: true
+        default: 'GIScience/openrouteservice'
+      branch:
+        description: 'GitHub branch to create image off.'
+        required: true
+        default: 'main'
+      tag:
+        description: 'Name of the docker tag to create.'
+        required: true
+        default: 'nightly'
+      ignore-24h-commit-check:
+        description: 'Build image regardless of last commit date.'
+        required: true
+        type: boolean
+        default: false
+      user:
+        description: ''
+        required: false
+        default: 'schedule'
+
+jobs:
+  check-commits:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4.1.1
+        with:
+          repository: ${{ github.event.inputs.repository || 'GIScience/openrouteservice' }}
+          ref: ${{ github.event.inputs.branch || 'main' }}
+
+      - name: Check for new commits since 24 h ago
+        run: echo "NEW_COMMIT_COUNT=$(git log --oneline --since '24 hours ago' | wc -l)" >> $GITHUB_ENV
+
+  build:
+    needs: check-commits
+    runs-on: ubuntu-latest
+    if: ${{ github.event.inputs.ignore-24h-commit-check || env.NEW_COMMIT_COUNT > 0 }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4.1.1
+        with:
+          repository: ${{ github.event.inputs.repository || 'GIScience/openrouteservice' }}
+          ref: ${{ github.event.inputs.branch || 'main' }}
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3.0.0
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3.0.0
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v3.0.0
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+
+      - name: Build and push
+        uses: docker/build-push-action@v5.1.0
+        with:
+          context: .
+          platforms: linux/amd64,linux/arm64/v8
+          provenance: false
+          push: true
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
```

Comment:
push true and cache-to together doesn't work, if i remember it correctly.

Reply:
do you have have more details on this?
as far as i understand the action docs that link to the docker [cache-to](https://docs.docker.com/engine/reference/commandline/buildx_build/#cache-to) for this and those examples there:
https://github.com/moby/buildkit?tab=readme-ov-file#github-actions-cache-experimental

it should work together?

Reply:
Hm. When I look it up, I cannot find it any more. I remember it in earlier versions being absolutely annoying about some combinations of parameters. I guess you tested it, so let's leave it if the action isn't complaining :)!